fenix/app/src/main/java/org/mozilla/fenix/translations/preferences/downloadlanguages/DownloadLanguageFileDialog.kt

```diff
@@ -0,0 +1,192 @@
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.defaultMinSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.selection.toggleable
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Checkbox
+import androidx.compose.material.CheckboxDefaults
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.clearAndSetSemantics
+import androidx.compose.ui.semantics.heading
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import mozilla.components.feature.downloads.toMegabyteOrKilobyteString
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.compose.button.PrimaryButton
+import org.mozilla.fenix.compose.button.TextButton
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Download Languages File Dialog.
+ *
+ * @param fileSize Language file size in bytes that should be displayed in the dialogue title.
+ * @param isCheckBoxEnabled Whether saving mode checkbox is checked or unchecked.
+ * @param isAllLanguagesItemType Whether the download language file item is of type all languages.
+ * @param onSavingModeStateChange Invoked when the user clicks on the checkbox of the saving mode state.
+ * @param onConfirmDownload Invoked when the user click on the "Download" dialog button.
+ * @param onCancel Invoked when the user clicks on the "Cancel" dialog button.
+ */
+@Suppress("LongParameterList")
+@Composable
+fun DownloadLanguageFileDialog(
+    fileSize: Long,
+    isCheckBoxEnabled: Boolean,
+    isAllLanguagesItemType: Boolean,
+    onSavingModeStateChange: (Boolean) -> Unit,
+    onConfirmDownload: () -> Unit,
+    onCancel: () -> Unit,
+) {
+    Dialog(onDismissRequest = {}) {
+        Column(
+            modifier = Modifier
+                .background(
+                    color = FirefoxTheme.colors.layer2,
+                    shape = RoundedCornerShape(8.dp),
+                )
+                .padding(16.dp),
+        ) {
+            Text(
+                text = stringResource(
+                    R.string.download_language_file_dialog_title,
+                    fileSize.toMegabyteOrKilobyteString(),
+                ),
+                modifier = Modifier
+                    .semantics { heading() },
+                color = FirefoxTheme.colors.textPrimary,
+                style = FirefoxTheme.typography.headline7,
+                fontWeight = FontWeight.Bold,
```

Comment:
@iorgamgabriel I am not seeing the need for `fontWeight = FontWeight.Bold` (this is equivalent to [FontWeight(700)](https://developer.android.com/reference/kotlin/androidx/compose/ui/text/font/FontWeight#Bold())) in the [figma](https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD/Full-Page-Translation?node-id=7202%3A96985&mode=dev). 

<img width="634" alt="Screen Shot 2023-12-27 at 1 57 36 PM" src="https://github.com/mozilla-mobile/firefox-android/assets/1190888/75616dc0-48e5-46fd-9e56-1b141d7664d8">